common/grammar-parser.cpp

```diff
@@ -188,6 +188,10 @@ namespace grammar_parser {
                     throw std::runtime_error(std::string("expecting ')' at ") + pos);
                 }
                 pos = parse_space(pos + 1, is_nested);
+            } else if (*pos == '.') { // any char
+                last_sym_start = out_elements.size();
+                out_elements.push_back({LLAMA_GRETYPE_CHAR_ANY, 0});
```

Comment:
Currently wondering if it's worth adding a whole new LLAMA_GRETYPE for "." just for this.

The alternative is that when we're parsing ".", instead of inserting `LLAMA_GRETYPE_CHAR_ANY` into `out_elements`, we could instead add TWO elements -- a `LLAMA_GRETYPE_CHAR` set to 0, followed immediately by a `LLAMA_GRETYPE_CHAR_RNG_UPPER` set to max. This should have the same functionality as `LLAMA_GRETYPE_CHAR_ANY`, but at the cost of having extra cruft in the data structure.

This would effectively change every "." into "[\x0000-\xffff]" (and be handled identically).

That feels like the RISC method, and the current implementation is the CISC method. I'm not sure which is better, but just reviewing and thinking out loud here. Now that I type it all out, I kinda' prefer the cleanliness of the `LLAMA_GRETYPE_CHAR_ANY` method -- even if we have to add a new entry to the enum.

Reply:
I would lean towards keeping things as simple as possible (just one more of many parser-rewritten artefacts), esp. since I doubt there's any measurable runtime difference between the two approaches.

Reply:
That said this doesn't add much complexity so also happy to see this merged as it is ;-)